src/app/(pages)/users/[id]/_components/UsersForm.tsx

```diff
@@ -0,0 +1,84 @@
+import { useForm } from "react-hook-form";
+import FormInput from "~/components/organisms/form/formInput/FormInput";
+import FormSwitch from "~/components/organisms/form/formSwitch/FormSwitch";
+import { Button } from "~/components/atoms/Button";
+import FormComponent from "~/components/organisms/form/formComponent/FormComponent";
+import React, { useState } from "react";
+
+type UserUpdateFormData = {
+  email: string;
+  active: boolean;
+  firstname: string;
+  lastname: string;
+};
+
+type Props = {
+  id: string;
+  formData: UserUpdateFormData;
+};
+
+const UserForm: React.FC<Props> = ({ id, formData }) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [active, setActive] = useState(formData.active);
+
+  const form = useForm<UserUpdateFormData>({
+    values: {
+      email: formData.email,
+      active: formData.active,
+      firstname: formData.firstname,
+      lastname: formData.lastname,
+    },
+  });
+
+  // Handle form submission
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    form.setValue("active", active);
+    setIsSubmitting(false);
```

Comment:
Could you check if we can use the native `isSubmitting` property from the forms?
https://react-hook-form.com/docs/useform/formstate search for `isSubmitting`

Reply:
It is working, great catch 🚀 
![image](https://github.com/user-attachments/assets/ae85f819-3908-4cdf-922f-0fb464defb6d)